### PR TITLE
Trait to test Inertia Controller

### PR DESCRIPTION
**Pull request by snake-py**
As we spoke [here](https://github.com/orchestral/toolkit-docs/pull/6#issuecomment-1482692370) I created a trait to test a controller, which response with inertia response. To use this you need to do the following:

1. Create a new file named app.blade.php in your package's resources/views directory.
2. Call the load function from the trait

```php

class FooTest  extends BaseTest
{
    use WithInertia;

    protected $inertiaMiddleware = HandleInertiaRequests::class;
}
```

You can customize the Finder using:

```php
use Illuminate\View\FileViewFinder;

class FooTest  extends BaseTest
{
    use WithInertia;

    protected $inertiaMiddleware = HandleInertiaRequests::class;

    public function defineEnvironment($app): void
    {
        $this->defineInertia(function (FileViewFinder $finder) {
            $finder->addNamespace('WD', __DIR__.'/../../UI/resources/app/Pages');
        });
    }
}
```

To define custom paths or register view location.

```php
$this->defineInertiaPages(view: 'resources/views', paths: ['resources/js/Pages']);

### CODE REVIEW

File: src/Concerns/WithInertiaController.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        array $fileFinderExtensions = ['vue', 'js', 'jsx', 'ts', 'tsx', 'html', 'php']
+    ) {
+        View::addLocation($viewPath);
+        $this->app->bind('inertia.testing.view-finder', function ($app) use ($pathToPages, $fileFinderExtensions) {
+            $viewFinder = new \Illuminate\View\FileViewFinder(
+                $app['files'],
+                array_merge($app['config']->get('inertia.testing.page_paths'), $pathToPages),
+                array_merge($app['config']->get('inertia.testing.page_extensions'), $fileFinderExtensions)
+            );
+            $viewFinder->addNamespace('WD', 'UI/resources/app/Pages');
```
Comment (crynobone) [inline src/Concerns/WithInertiaController.php:22]: This shouldn't hardcoded.

**Review by crynobone — CHANGES_REQUESTED** [review]

File: src/Concerns/WithInertiaController.php (hunk 1 above)
Comment (snake-py) [reply src/Concerns/WithInertiaController.php]: You are right sorry this slipped through. I made this as an optional param.

Comment (snake-py) [reply src/Concerns/WithInertiaController.php]: @crynobone Will I also need to add an example or is it fine like this?

File: src/Concerns/WithInertia.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+     * @param  (callable(\Illuminate\View\FileViewFinder, \Illuminate\Foundation\Application, \Illuminate\Contracts\Config\Repository):void)|null  $callback
+     * @param  bool|null  $ensurePageExists
+     * @return void
+     */
+    protected function defineInertia(callable $callback = null): void
+    {
+        /** @var \Illuminate\Contracts\Config\Repository $config */
+        $config = $this->app['config'];
+
+        $this->app['config']->set('inertia.testing.ensure_pages_exist', false);
```
Comment (crynobone) [inline src/Concerns/WithInertia.php:42]:
```suggestion
        $config->set('inertia.testing.ensure_pages_exist', false);
```

**Review by crynobone — APPROVED** [review]

**Comment by snake-py** [comment]
@crynobone I am unsure if I can assist here in any way. I see that you split up the code. SHould I try to tackle this or do you want to do it?

**Pull request closed without merging**